[export] Disable exported_program.__call__

angelayi opened the pull request:
Summary: `ExportedProgram` is an artifact produced by torch.export, containing the graph that is exported, along with other attributes about the original program such as the graph signature, state dict, and constants. One slightly confusing thing that users run into is that they treat the `ExportedProgram` as a `torch.nn.Module`, since the object is callable. However, as we do not plan to support all features that `torch.nn.Module`s have, like hooks, we want to create a distinction between it and the `ExportedProgram` by removing the `__call__` method. Instead users can create a proper `torch.nn.Module` through `exported_program.module()` and use that as a callable.

Test Plan: CI

Differential Revision: D53075378




cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @chenyang78 @aakhundov @kadeng

tugsbayasgalan commented on torch/export/exported_program.py:
```diff
@@ -275,6 +275,11 @@ def constants(self):
+            "You should use `exported_program.module()` instead."
+        )
+
         import torch._export.error as error
```
Delete all the below code?

angelayi committed:
[export] Disable exported_program.__call__ (#119466)

Summary:
X-link: https://github.com/pytorch/executorch/pull/1954


`ExportedProgram` is an artifact produced by torch.export, containing the graph that is exported, along with other attributes about the original program such as the graph signature, state dict, and constants. One slightly confusing thing that users run into is that they treat the `ExportedProgram` as a `torch.nn.Module`, since the object is callable. However, as we do not plan to support all features that `torch.nn.Module`s have, like hooks, we want to create a distinction between it and the `ExportedProgram` by removing the `__call__` method. Instead users can create a proper `torch.nn.Module` through `exported_program.module()` and use that as a callable.

Test Plan: CI

Reviewed By: zhxchen17

Differential Revision: D53075378

angelayi commented:
cc @peri044 @qihqi

huydhn commented:
Dr.CI treats the failures as a flaky ones because it detects similar ones elsewhere https://hud.pytorch.org/failure?name=pull%20%2F%20linux-focal-py3.8-clang10-onnx%20%2F%20test%20(default%2C%201%2C%202%2C%20linux.2xlarge)&jobName=linux-focal-py3.8-clang10-onnx%20%2F%20test%20(default%2C%201%2C%202%2C%20linux.2xlarge)&failureCaptures=%5B%22onnx%2Ftest_fx_to_onnx_with_onnxruntime.py%3A%3ATestFxToOnnxWithOnnxRuntime_op_level_debug_True_dynamic_shapes_True_model_type_TorchModelType.TORCH_NN_MODULE%3A%3Atest_add%22%5D but in this case it is a false positive.  The failure is not flaky
